traffic_monitor/experimental/common/poller/poller.go

```diff
@@ -157,68 +187,151 @@ func (p HttpPoller) Poll() {
 			}
-			go pollHttp(info.Interval, info.ID, info.URL, fetcher, kill)
+			go sleepPoller(info.Interval, info.ID, info.URL, fetcher, kill)
 		}
 		p.Config = newConfig
 	}
 }
 
-type HTTPPollInfo struct {
-	Interval time.Duration
-	Timeout  time.Duration
-	ID       string
-	URL      string
-	Handler  handler.Handler
+func mustDie(die <-chan struct{}) bool {
+	select {
+	case <-die:
+		return true
+	default:
+	}
+	return false
 }
 
-// diffConfigs takes the old and new configs, and returns a list of deleted IDs, and a list of new polls to do
-func diffConfigs(old HttpPollerConfig, new HttpPollerConfig) ([]string, []HTTPPollInfo) {
-	deletions := []string{}
-	additions := []HTTPPollInfo{}
+// TODO iterationCount and/or p.TickChan?
+func sleepPoller(interval time.Duration, id string, url string, fetcher fetcher.Fetcher, die <-chan struct{}) {
+	pollSpread := time.Duration(rand.Float64()*float64(interval/time.Nanosecond)) * time.Nanosecond
+	time.Sleep(pollSpread)
+	tick := time.NewTicker(interval)
+	lastTime := time.Now()
+	for {
+		select {
+		case <-tick.C:
+			realInterval := time.Now().Sub(lastTime)
+			if realInterval > interval+(time.Millisecond*100) {
+				instr.TimerFail.Inc()
+				log.Infof("Intended Duration: %v Actual Duration: %v\n", interval, realInterval)
+			}
+			lastTime = time.Now()
 
-	if old.Interval != new.Interval {
-		for id, _ := range old.Urls {
-			deletions = append(deletions, id)
+			pollId := atomic.AddUint64(&debugPollNum, 1)
+			pollFinishedChan := make(chan uint64)
+			log.Debugf("poll %v %v start\n", pollId, time.Now())
+			go fetcher.Fetch(id, url, pollId, pollFinishedChan) // TODO persist fetcher, with its own die chan?
+			<-pollFinishedChan
+		case <-die:
+			return
 		}
-		for id, pollCfg := range new.Urls {
-			additions = append(additions, HTTPPollInfo{
-				Interval: new.Interval,
+	}
+}
+
+// InsomniacPoll polls using a single thread, which never sleeps. This exists to work around a bug observed in OpenStack CentOS 6.5 kernel 2.6.32 wherin sleep gets progressively slower. This should be removed and Poll() changed to call SleepPoll() when the bug is tracked down and fixed for production.
+func (p HttpPoller) InsomniacPoll() {
+	// iterationCount := uint64(0)
+	// iterationCount++ // on tick<:
+	// case p.TickChan <- iterationCount:
+	killChan := make(chan struct{})
+	pollRunning := false // TODO find less awkward way to not kill the first loop
+	pollerId := rand.Int63()
+	for newCfg := range p.ConfigChannel {
+		// TODO add a more efficient function than diffConfigs for this func, since we only need to know whether anything changed
+		deletions, additions := diffConfigs(p.Config, newCfg)
+		if len(deletions) == 0 && len(additions) == 0 {
+			continue
+		}
+
+		fmt.Printf("HttpPoller.InsomniacPoll got newCfg\n")
+		if pollRunning {
+			killChan <- struct{}{}
+		}
+		pollRunning = true
+
+		polls := []HTTPPollInfo{}
+		fmt.Printf("HttpPoller.InsomniacPoll creating polls\n")
```

Comment:
Looks like there are several `fmt.Printf`s so I will just comment once here.  Are they all necessary or are they left over from debugging?  If they are necessary, maybe they should be logs instead of prints?